core/store/orm/orm.go

```diff
@@ -614,29 +614,33 @@ func (orm *ORM) AnyJobWithType(taskTypeName string) (bool, error) {
 	return found, ignoreRecordNotFound(rval)
 }
 
-// CreateTx returns a transaction by its surrogate key, if it exists, or
+// CreateTx finds and overwrites a transaction by its surrogate key, if it exists, or
 // creates it
 func (orm *ORM) CreateTx(tx *models.Tx) (*models.Tx, error) {
 	orm.MustEnsureAdvisoryLock()
 
 	err := orm.convenientTransaction(func(dbtx *gorm.DB) error {
 		var query *gorm.DB
+		foundTx := models.Tx{}
 		if tx.SurrogateID.Valid {
-			query = dbtx.First(&models.Tx{}, "surrogate_id = ?", tx.SurrogateID.ValueOrZero())
+			query = dbtx.First(&foundTx, "surrogate_id = ?", tx.SurrogateID.ValueOrZero())
+			// TODO: Put ON CONFLICT REPLACE ALL here
+			// https://github.com/sljeff/gorm/pull/5
 		} else {
-			query = dbtx.First(&models.Tx{}, "hash = ?", tx.Hash)
+			// TODO: Make hash unique so we can ON CONFLICT REPLACE ALL here
+			query = dbtx.First(&foundTx, "hash = ?", tx.Hash)
 		}
+		err := query.Error
 
-		ids := []uint64{}
-		err := query.Pluck("id", &ids).Error
```

Comment:
This query does a literal `select id from txes;`.

It always returns every single ID in the database and puts that into `ids` regardless of filters on query.

Reply:
This seems to contradict the gorm documentation and examples - I wonder if it's a bug there?

Reply:
I suspect it may be a bug. Went to open an issue but I saw that he is currently in the process of rewriting most of this logic into gorm 2.0 so didn't figure it was worth it.